pandas/core/_meta.py

```diff
@@ -0,0 +1,146 @@
+"""
+Metadata propagation through pandas operations.
+
+This module contains the infrastructure for propagating ``NDFrame._metadata``
+through operations. We perform an operation (say :meth:`pandas.Series.copy`) that
+returns an ``NDFrame`` and would like to propagate the metadata (say ``Series.name``)
+from ``self`` to the new ``NDFrame``.
+
+.. note::
+
+   Currently, pandas doesn't provide a clean, documented API on
+
+   * which methods call finalize
+   * the types passed to finalize for each method
+
+   This is a known limitation we would like to address in the future.
+"""
+from collections import defaultdict
+from functools import wraps
+from typing import TYPE_CHECKING, Any, Callable, Union
+
+from pandas.core.dtypes.generic import ABCDataFrame, ABCSeries
+
+if TYPE_CHECKING:
+    from pandas.core.generic import NDFrame
+
+dispatch = defaultdict(dict)
```

Comment:
Can you annotate this? Would help read what should be nested therein

Reply:
I *think* it's supposed to be `DefaultDict[dispatch_method_type, Callable[..., FrameOrSeries]]`